src/config.xml

```diff
@@ -994,6 +994,17 @@ Go to the <a href="commands.html">next</a> section or return to the
  various overviews. This option will also hide undocumented C++ concepts
  if enabled.
  This option has no effect if \ref cfg_extract_all "EXTRACT_ALL" is enabled.
+]]>
+      </docs>
+    </option>
+    <option type='bool' id='HIDE_UNDOC_NAMESPACES' defval='1'>
```

Comment:
I think defval should be 0 for backward compatible behavior.

Reply:
I don't think so.
The key line here is:
```
(hasDocumentation() || !hideUndoc || getLanguage()==SrcLangExt::CSharp)
```
where the `|| !hideUndoc` has been added and has the value `bool hideUndoc = Config_getBool(HIDE_UNDOC_NAMESPACES);`.

So in the default value case of `1` (`YES`) the `!hideUndoc`  is `false` and does not alter the value of the `||` construct.

Reply:
Indeed, I was confused by the default of `HIDE_UNDOC_CLASSES` being `1` and the double/tripple negation logic used. i.e. `!hideUndoc`, which in my head I incorrectly interpreted as`showDoc` instead of `showUndoc`.